opm/flowdiagnostics/utility/AssembledConnections.cpp

```diff
@@ -256,9 +256,7 @@ CSR::accumulateRowEntries(const int               numRows,
     }
 
     // Note index range: 1..numRows inclusive.
-    for (decltype(this->ia_.size())
-             i = 1, numRows = this->ia_.size() - 1;
-         i <= numRows; ++i)
+    for (int i = 1; i <= numRows; ++i)
```

Comment:
By changing the type of `i` from `decltype(ia_.size())` (typically `std::size_t`) to `int` you're incurring a type conversion on every iteration (possibly every reference to `std::vector<>::operator[]()`.  Please change it back and rename the upper bound.